core/cap-0034.md

```diff
@@ -0,0 +1,352 @@
+# CAP-0034: Preserve Transaction-Set/Close-Time Affinity During Nomination
+
+## Preamble
+
+```text
+CAP: To Be Assigned
+Title: Preserve Transaction-Set/Close-Time Affinity During Nomination
+Authors: Terence Rokop, Nicolas Barry
+Status: Draft
+Created: 2020-07-06
+Discussion: https://github.com/stellar/stellar-protocol/issues/622
+Protocol version: TBD
+```
+
+## Simple Summary
+
+This CAP proposes to change the way that the nomination protocol generates
+a `closeTime` to externalize -- instead of taking the maximum `closeTime` of
+all candidates, as in current protocols, it proposes to use the specific
+`closeTime` in the same `StellarValue` as the chosen transaction set. It
+thereby closes a race which can allow a transaction to pay a fee and consume
+a sequence number, yet fail before being applied as having committed "too late".
+This CAP also proposes to preserve the signature on that `StellarValue`
+throughout the nomination, as the `closeTime` change allows this.
+
+## Motivation
+
+With the current protocol, a transaction with an expiration time might make it
+into a candidate `StellarValue`'s transaction set, with its expiration being no
+later than the `closeTime` in that `StellarValue`, but then fail as "too late"
+during ledger close, as the transaction applies during ledger close happen
+after the "current" ledger header, including the `closeTime`, has been updated
+to reflect the externalized `StellarValue`.
+
+When that happens, a transaction is charged a fee, and consumes a sequence
+number, despite having been delayed long enough to exceed its expiration time.
+The fee and sequence number aspects are potentially both problems:
+
+- Transactions submitted by SDKs generally set expiration times so that they
+know when it's safe to re-submit them (with new sequence numbers and perhaps
+higher fees) if they don't seem to be making it into transaction sets with their
+current offered fees (or because of transient network congestion or memory
+constraints on a validator).  It would make more sense if they were only charged
+the fee that they were willing to pay that actually got them accepted into a
+transaction set.
+
+- Probably worse, there is a design pattern in some smart contracts which
```

Comment:
Alright, I've tried to avoid saying "fail" or "invalid" when I meant the other.

Yeah, I noticed that `B` didn't end up getting used.  I thought I had an idea of how it would be, but it turned out superfluous.  I've removed it (as well as the speculation about expiration duration and consequent rarity).